### DISCUSSION TITLE
Access Other Fields in Required Validation Closure

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#1386] [Help] [answered]
I know that the `required()` rule can accept a closure, like this example from the docs:

```
TextInput::make('middle_name')
    ->required(function () {
        return auth()->user()->hasMiddleName();
    })
```

But how would I access another field being validated, from within that closure? My use case is that I only want to require a state if the selected country is the US. I tried simply using the `required_if` rule, like the code below shows, but it is not working for me for some reason.

```
->rules(['required_if:country,US'])
```

**Comment by intrepidws** [comment]
Well, of course I figure it out basically as soon as I post this. I was able to get the following code to work for a `required_if` validation.

```
->rules(['required_if:data.country,US'])
```

**Comment by intrepidws** [comment, marked as answer]
I'd still love to know how to access other fields from within the closure though.

Edit: Dan has provided the answer below.

**Reply by danharrin** [reply]
You can access another fields value using `$get('field_name')` in any closure. So:

`->required(fn ($get) => $get('country') === 'US')`

https://filamentadmin.com/docs/2.x/forms/advanced#using-closure-customisation

**Reply by intrepidws** [reply]
Thanks for your help here. I reviewed that exact page of the documentation, I don't know how I missed that.
